static/app/components/group/groupSummary.spec.tsx

```diff
@@ -100,4 +102,174 @@ describe('GroupSummary', function () {
+            {
+              provider: 'integrations:github',
+              owner: 'getsentry',
+              name: 'sentry',
+              external_id: '123',
+            },
+          ],
+        },
+      },
+    });
+
+    render(<GroupSummary groupId={groupId} groupCategory={IssueCategory.PERFORMANCE} />);
+
+    expect(screen.queryByText('Issue Summary')).not.toBeInTheDocument();
+    expect(screen.queryByText('Test summary')).not.toBeInTheDocument();
+    expect(screen.queryByText('Potential Impact')).not.toBeInTheDocument();
+    expect(screen.queryByText('Test impact')).not.toBeInTheDocument();
+  });
+});
+
+describe('GroupSummaryHeader', function () {
+  beforeEach(() => {
+    MockApiClient.clearMockResponses();
+  });
+
+  it('renders the group summary header', async function () {
+    const groupId = '1';
+    const organizationSlug = 'org-slug';
+    MockApiClient.addMockResponse({
+      url: makeGroupSummaryQueryKey(organizationSlug, groupId)[0],
+      method: 'POST',
+      body: {
+        groupId,
+        summary: 'Test summary',
+        impact: 'Test impact',
+        headline: 'Test headline',
+      },
+    });
+
+    MockApiClient.addMockResponse({
+      url: `/issues/${groupId}/autofix/setup/`,
+      body: {
+        genAIConsent: {ok: true},
+        integration: {ok: true},
+        githubWriteIntegration: {
+          ok: true,
+          repos: [
+            {
+              provider: 'integrations:github',
+              owner: 'getsentry',
+              name: 'sentry',
+              external_id: '123',
+            },
+          ],
+        },
+      },
+    });
+
+    render(<GroupSummaryHeader groupId={groupId} groupCategory={IssueCategory.ERROR} />);
+
+    await waitFor(() => {
+      expect(screen.getByText('Test headline')).toBeInTheDocument();
+    });
+  });
+
+  it('does not render the group summary headline if no consent', async function () {
+    const groupId = '1';
+    const organizationSlug = 'org-slug';
+    MockApiClient.addMockResponse({
+      url: makeGroupSummaryQueryKey(organizationSlug, groupId)[0],
+      method: 'POST',
+      body: {
+        groupId,
+        summary: 'Test summary',
+        impact: 'Test impact',
+        headline: 'Test headline',
+      },
+    });
+
+    const setupCall = MockApiClient.addMockResponse({
+      url: `/issues/${groupId}/autofix/setup/`,
+      body: {
+        genAIConsent: {ok: false},
+        integration: {ok: true},
+        githubWriteIntegration: {
+          ok: true,
+          repos: [
+            {
+              provider: 'integrations:github',
+              owner: 'getsentry',
+              name: 'sentry',
+              external_id: '123',
+            },
+          ],
+        },
+      },
+    });
+
+    render(<GroupSummaryHeader groupId={groupId} groupCategory={IssueCategory.ERROR} />);
+
+    await waitFor(
+      () => {
+        expect(setupCall).toHaveBeenCalled();
+      },
+      {timeout: 5000}
+    );
```

Comment:
another tiny nit but i would hope this call happens instantly not within 5 seconds
```suggestion
    await waitFor(
      () => {
        expect(setupCall).toHaveBeenCalled();
      }
    );
```